third_problem.rb

```diff
@@ -1,13 +1,68 @@
-
 class ThirdProblem
   def initialize(numbers_array)
     @numbers = numbers_array
   end
 
   def hasDuplicates
-    if @numbers.empty?||@numbers.length==1 then
+    if @numbers.empty? || @numbers.length == 1 then
       return false
     end
+    duplicate = false
+
+    for it in 0...@numbers.length
```

Comment:
Just a thing to remember:
We write code for humans, not machines. It is very difficult to follow the code if the naming is not self-descriptive. In this case one has to do the association `it` = `index`

Reply:
For an iterator between 0 and x I would consider "it" as reasonable enough for a simple use-case like this . However when code gets more complex i agree it's worth to put a better name.
Would you prefer that i change this to index ? I personally don't see that much the added value.